jconstraints-core/src/main/java/gov/nasa/jpf/constraints/smtlibUtility/SMTProblem.java

```diff
@@ -55,4 +56,9 @@ public SolverContext addProblemToContext(SolverContext ctx) {
     }
     return ctx;
   }
+
+  public void addFunction(Function fct) {
+    assert !this.functions.containsKey(fct.getName());
```

Comment:
Do we need the performance savings of assert here? Otherwise, we could always throw `IllegalAlrgumentException` or similar

```suggestion
    if (this.functions.containsKey(fct.getName())) throw new IllegalArgumentException("duplicate key: " + fct.getName());
```

Reply:
Taken from the [SMT-Lib definition for declare-fun](https://smt-lib.org/papers/smt-lib-reference-v2.6-r2024-09-20.pdf)

```The command reports an error if a function symbol with name f is already
present in the current signature.```

In consequence, I think this must be an if.